lib/dpul_collections/indexing_pipeline/figgy/hydration_cache_entry.ex

```diff
@@ -32,10 +32,38 @@ defmodule DpulCollections.IndexingPipeline.Figgy.HydrationCacheEntry do
       years_is: extract_years(data),
       display_date_s: format_date(metadata),
       page_count_i: page_count(metadata),
-      image_service_urls_ss: image_service_urls(metadata, related_data)
+      image_service_urls_ss: image_service_urls(metadata, related_data),
+      primary_thumbnail_service_url_s: primary_thumbnail_service_url(metadata, related_data)
     }
   end
 
+  defp primary_thumbnail_service_url(
```

Comment:
I don't know if it's more readable, but I wonder if a bunch of these conditionals are pattern matchable?

Like

```elixir
defp primary_thumbnail_service_url(%{"thumbnail_id" => [%{"id" => id}], %{"member_ids" => %{id => stuff}})
```
for the case where there's an id key (this might not be possible, I dunno if you can match like this...)

Reply:
I tried similar approaches and could not get matches to happen.